src/rebar_pkg_resource.erl

```diff
@@ -210,15 +210,16 @@ store_etag_in_cache(Path, ETag) ->
       UpdateETag :: boolean(),
       Res :: ok | {unexpected_hash, integer(), integer()} | {fetch_fail, binary(), binary()}
            | {bad_registry_checksum, integer(), integer()} | {error, _}.
-cached_download(TmpDir, CachePath, Pkg={pkg, Name, Vsn, _OldHash, _Hash, RepoConfig}, State, ETag,
+cached_download(TmpDir, CachePath, Pkg={pkg, Name, Vsn, _OldHash, _Hash, RepoConfig}, _State, ETag,
                 ETagPath, UpdateETag) ->
-    CDN = maybe_default_cdn(State),
-    case request(RepoConfig#{repo_url => CDN}, Name, Vsn, ETag) of
+    RepoUrl = maps:get(repo_url, RepoConfig, undefined),
+    ?DEBUG("Downloading package ~ts from repo ~ts", [Name, RepoUrl]),
```

Comment:
Same comment review here about the debug message. This isn't accurate and can be misleading. I understand you want to see the hex cdn being active, but this isn't a good way to get it because for other issues (i.e. people configuring the hex repo config without that URL), it actually obscures the information that's going on.

Do note that the package may also not be downloaded, but just fetched from cache. The debug call about downloading should happen nearer to the actual download call.

Reply:
I agreed with you about this. `repo_url` isn't the actual requesting URL but only a prefix endpoint. I think we can debug `repo_url` right before calling `r3_hex_repo:some_function` which do the actual request. 

But here is the appropriate place to print debug info I think, because the next `request` call do the actual http request to the tarball via `r3_hex_repo:get_tarball`. And more, the next debug info right after the `request` call will show the actual place the tarball got from, which was added by me.

```erlang
        {ok, Body, NewETag} ->
            ?DEBUG("Downloaded package from repo ~ts, caching at ~ts", [CDN, CachePath]),
```

I notice that the `RepoConfig#{repo_url := URI}` was used by the `build_url` in `vendored/r3_hex_repo.erl`, which means `repo_url` here must have a value.

The hard thing to me is how to describe this debug info, `Making request to the repo hexpm via repo_url https://xxx.com/hexpm`, does this acceptable?